pyproject.toml

```diff
@@ -0,0 +1,101 @@
+[project]
+# https://packaging.python.org/en/latest/specifications/declaring-project-metadata/
+name = "cm-service"
+description = "Rubin Observatory campaign management FastAPI service"
+license = { file = "LICENSE" }
+readme = "README.md"
+keywords = ["rubin", "lsst"]
+# https://pypi.org/classifiers/
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.10",
+    "Intended Audience :: Developers",
+    "Natural Language :: English",
+    "Operating System :: POSIX",
+    "Typing :: Typed",
+]
+requires-python = ">=3.10"
+# Use requirements/main.in for runtime dependencies instead.
+dependencies = []
+dynamic = ["version"]
+
+[project.scripts]
+cm-service = "lsst.cmservice.cli.commands:main"
+
+[project.urls]
+Homepage = "https://cm-service.lsst.io"
+Source = "https://github.com/lsst-dm/cm-service"
+
+[build-system]
+requires = ["setuptools>=61", "wheel", "setuptools_scm[toml]>=6.2"]
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools_scm]
```

Comment:
Yes, I am planning semantic versioning for this; the ReST API to the service itself would be the driver.

Playing with this a little as a result of your question, I realize I'm a little unclear on the heuristics the scm plugin actually uses when configured by default this way.  I'll poke at it a little more and make sure I understand how it is working before merging.

Reply:
Okay, without configuration this derives from tags in `git` relative to the directory containing `pyproject.toml`.  I dropped a `0.0.0` tag on the repository initial empty commit, and this seems to be generating sensible results.

Reply:
Not explicitly, though `importlib.metadata.version()` does work correctly (for example, click's `@version-option` decorator just works as expected.

I could add some code to the package top-level `__init__` to call `importlib.metadata.version()` and explicitly set `lsst.cmservice.__version__` if you think this is desirable?

Reply:
I'm sure it's fine then. setuptools_scm has an option to write the version number into a file but maybe the future for everyone is `importlib.metadata` rather than using `__version__`. When you `pydoc lsst.cmservice` does it fill in a version at the bottom?

Reply:
Let me give that a shot...

Reply:
Good catch -- apparently not, unless I add the explicit `importlib.metadata` -> `__version__` plumbing to the top-level `__init__`.  Maybe `pydoc` hasn't caught up to `importlib` yet? In any case, I'll add the plumbing for now.

Reply:
or you get setuptools_scm to write a version file and then import it without asking importlib to do it every time. @rra is there a convention for this these days?

Reply:
For `__init__.py`:
```python
__all__ = ["__version__"]

from importlib.metadata import PackageNotFoundError, version

__version__: str
"""The application version string (PEP 440 / SemVer compatible)."""

try:
    __version__ = version("example")
except PackageNotFoundError:
    # package is not installed
    __version__ = "0.0.0"
```

Reply:
Thanks @rra -- that's essentially what I was looking at.  I might experiment with the file-writing config option as well...

Reply:
Oh, I see, I guess I'd need at least one line in `__init__.py` to import the written file anyway, so I'll just go with the above.